docs/concepts/controller-manager.md

```diff
@@ -72,4 +72,54 @@ The `--event-ttl` should be larger than the `ttlNonShootEvents` or this controll
+
+### Kubeconfig Secret Validator
+
+[Malicious Kubeconfigs](https://github.com/kubernetes/kubectl/issues/697) applied by end users may cause a leakage of sensitive data.
+This handler checks if the incoming request contains a Kubernetes secret with a `.data.kubeconfig` field and denies the request if the Kubeconfig structure violates Gardener's security standards.
+
+### Namespace Validator
+
+Namespaces are the backing entities of Gardener projects in which shoot clusters objects reside.
+This validation handler protects active namespaces against premature deletion requests.
+Therefore, it denies deletion requests if a namespace still contains shoot clusters or if it belongs to a non-deleting Gardener project (project w/o `.meta.deletionTimestamp`).
+
+### Resource Size Validator
+
+Since users directly apply Kubernetes native objects to the Garden cluster, it also involves the risk of being susceptible to DoS attacks because these resources are again read by further controllers in the Gardener landscape.
+One example is the creation of shoot clusters with large annotation values (up to 256 kB per value) which can cause severe out-of-memory issues for the Gardenlet component.
+[Vertical autoscaling](https://github.com/kubernetes/autoscaler/tree/master/vertical-pod-autoscaler) can help to mitigate such situations, but we cannot expect to scale infinitely and thus need means to block the attack itself.
+
+The Resource Size Validator checks arbitrary incoming admission requests against a configured maximum size for the resource's group-version-kind combination and denies the request if the contained object exceeds the quota.
+
+Example for Gardener Controller Manager configuration:
+```
+server:
+  resourceAdmissionConfiguration:
+    limits:
+    - apiGroups: ["core.gardener.cloud"]
+      apiVersions: ["*"]
+      resources: ["shoots", "plants"]
+      size: 100k
+    - apiGroups: [""]
+      apiVersions: ["v1"]
+      resources: ["secrets"]
+      size: 100k
+    unrestrictedUsers:
+    - groupName: "gardener.cloud:system:seeds"
+      userNames: ["*"]
+    operationMode: block #log
+```
+
+With the configuration above, the Resource Size Validator denies requests for shoots and plants with Gardener's core API group which exceed a size of 100 kB. The same is done for Kubernetes secrets.
+
+As this feature is meant to protect the system from malicious requests sent by users, it is recommended to exclude trusted groups and users from the size restriction via `resourceAdmissionConfiguration.unrestrictedUsers`.
+For example, the backing user for the Gardenlet should always be capable of changing the shoot resource instead of being the one that locks itself out due to size restrictions.
+This is because the Gardenlet itself occasionally changes the shoot specification, labels or annotations and might violate against the quota if the existing resource is already close to the quota boundary.
```

Comment:
```suggestion
This is because the Gardenlet itself occasionally changes the shoot specification, labels or annotations, and might violate the quota if the existing resource is already close to the quota boundary.
```